service/controller/app/v1/resource/status/create.go

```diff
@@ -36,15 +37,22 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found status for chart %#q", name))
 
-	if chart.Status.Status != "" && key.AppStatus(cr) != chart.Status.Status {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting app %#q status as %#q", name, chart.Status.Status))
+	chartStatus := key.ChartStatus(*chart)
+	desiredStatus := v1alpha1.AppStatus{
+		AppVersion: chartStatus.AppVersion,
+		Release: v1alpha1.AppStatusRelease{
+			LastDeployed: *chartStatus.Release.LastDeployed.DeepCopy(),
+			Status:       chartStatus.Release.Status,
+		},
+		Version: chartStatus.Version,
+	}
```

Comment:
Approach is updated a bit so we calculate the desired state and then use the `equals` func for comparison.  This means all fields are compared.